File: app-old/app/src/main/java/org/jesperancinha/universitybaseconverter/UniversityBaseDecToBaseActivity.java
package nl.joaofilipesabinoesperancinha.basicunitconverter;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.GestureDetector;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuItem;
import android.view.MotionEvent;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

import nl.joaofilipesabinoesperancinha.basicunitconverter.R;

public class UniversityBaseDecToBaseActivity extends AppCompatActivity implements GestureDetector.OnGestureListener {
    private static final int SWIPE_MIN_DISTANCE = 50;
    private static final int SWIPE_MAX_OFF_PATH = 200;
    private static final int SWIPE_THRESHOLD_VELOCITY = 200;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_university_base_dec_to_base);

        if (savedInstanceState == null) {
            getSupportFragmentManager().beginTransaction()
                    .add(R.id.container, new PlaceholderFragment())
                    .commit();
        }
    }


    @Override
    public boolean onCreateOptionsMenu(Menu menu) {

        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.university_base_dec_to_base, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();
        if (id == R.id.action_settings) {
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    /**
     * A placeholder fragment containing a simple view.
     */
    public static class PlaceholderFragment extends Fragment {

        public PlaceholderFragment() {
        }

        @Override
        public View onCreateView(LayoutInflater inflater, ViewGroup container,
                                 Bundle savedInstanceState) {
            View rootView = inflater.inflate(R.layout.fragment_university_base_dec_to_base, container, false);
            return rootView;
        }
    }

    @Override
    public boolean onDown(MotionEvent motionEvent) {
        return false;
    }

    @Override
    public void onShowPress(MotionEvent motionEvent) {

    }

    @Override
    public boolean onSingleTapUp(MotionEvent motionEvent) {
        return false;
    }

    @Override
    public boolean onScroll(MotionEvent motionEvent, MotionEvent motionEvent2, float v, float v2) {
        return false;
    }

    @Override
    public void onLongPress(MotionEvent motionEvent) {

    }

    @Override
    public boolean onFling(MotionEvent e1, MotionEvent e2, float velocityX,
                           float velocityY) {
        try {
            Toast t = Toast.makeText(UniversityBaseDecToBaseActivity.this, "Gesture detected", Toast.LENGTH_SHORT);
            t.show();
            float diffAbs = Math.abs(e1.getY() - e2.getY());
            float diff = e1.getX() - e2.getX();

            if (diffAbs > SWIPE_MAX_OFF_PATH)
                return false;

            // Left swipe
            if (diff > SWIPE_MIN_DISTANCE
                    && Math.abs(velocityX) > SWIPE_THRESHOLD_VELOCITY) {
                UniversityBaseDecToBaseActivity.this.onLeftSwipe();
            }
            // Right swipe
            else if (-diff > SWIPE_MIN_DISTANCE
                    && Math.abs(velocityX) > SWIPE_THRESHOLD_VELOCITY) {
                UniversityBaseDecToBaseActivity.this.onRightSwipe();
            }
        } catch (Exception e) {
            Log.e("Home", "Error on gestures");
        }
        return false;
    }

    private void onLeftSwipe() {
        Toast t = Toast.makeText(UniversityBaseDecToBaseActivity.this, "Left swipe", Toast.LENGTH_LONG);
        t.show();
        Intent go = new Intent(UniversityBaseDecToBaseActivity.this, UniversityBaseBaseToDecActivity.class);
        finish();
        startActivity(go);
    }

    private void onRightSwipe() {
        Toast t = Toast.makeText(UniversityBaseDecToBaseActivity.this, "Right swipe", Toast.LENGTH_LONG);
        t.show();
        Intent go = new Intent(UniversityBaseDecToBaseActivity.this, UniversityBaseBaseToDecActivity.class);
        finish();
        startActivity(go);
    }

}

File: app/src/main/java/nl/joaofilipesabinoesperancinha/basicunitconverter/UniversityBaseDecToBaseActivity.kt
package nl.joaofilipesabinoesperancinha.basicunitconverter

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.GestureDetector
import android.view.LayoutInflater
import android.view.Menu
import android.view.MenuItem
import android.view.MotionEvent
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import kotlin.math.abs

class UniversityBaseDecToBaseActivity : AppCompatActivity(), GestureDetector.OnGestureListener {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_university_base_dec_to_base)

        if (savedInstanceState == null) {
            getSupportFragmentManager().beginTransaction()
                .add(R.id.container, PlaceholderFragment())
                .commit()
        }
    }


    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.

        getMenuInflater().inflate(R.menu.university_base_dec_to_base, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        val id = item.itemId
        if (id == R.id.action_settings) {
            return true
        }
        return super.onOptionsItemSelected(item)
    }

    /**
     * A placeholder fragment containing a simple view.
     */
    class PlaceholderFragment : Fragment() {
        override fun onCreateView(
            inflater: LayoutInflater, container: ViewGroup?,
            savedInstanceState: Bundle?
        ): View? {
            val rootView: View =
                inflater.inflate(R.layout.fragment_university_base_dec_to_base, container, false)
            return rootView
        }
    }

    override fun onDown(motionEvent: MotionEvent): Boolean {
        return false
    }

    override fun onShowPress(motionEvent: MotionEvent) {
    }

    override fun onSingleTapUp(motionEvent: MotionEvent): Boolean {
        return false
    }

    override fun onScroll(
        motionEvent: MotionEvent?,
        motionEvent2: MotionEvent,
        v: Float,
        v2: Float
    ): Boolean {
        return false
    }

    override fun onLongPress(motionEvent: MotionEvent) {
    }

    override fun onFling(
        e1: MotionEvent?,
        e2: MotionEvent,
        velocityX: Float,
        velocityY: Float
    ): Boolean {
        try {
            val t: Toast = Toast.makeText(
                this@UniversityBaseDecToBaseActivity,
                "Gesture detected",
                Toast.LENGTH_SHORT
            )
            t.show()
            val diffAbs: Double = abs((requireNotNull(e1).y - e2.y).toDouble())
            val diff: Float = e1.x - e2.x

            if (diffAbs > SWIPE_MAX_OFF_PATH) return false

            // Left swipe
            if (diff > SWIPE_MIN_DISTANCE
                && abs(velocityX.toDouble()) > SWIPE_THRESHOLD_VELOCITY
            ) {
                this@UniversityBaseDecToBaseActivity.onLeftSwipe()
            } else if (-diff > SWIPE_MIN_DISTANCE
                && abs(velocityX.toDouble()) > SWIPE_THRESHOLD_VELOCITY
            ) {
                this@UniversityBaseDecToBaseActivity.onRightSwipe()
            }
        } catch (e: Exception) {
            Log.e("Home", "Error on gestures")
        }
        return false
    }

    private fun onLeftSwipe() {
        val t: Toast =
            Toast.makeText(this@UniversityBaseDecToBaseActivity, "Left swipe", Toast.LENGTH_LONG)
        t.show()
        val go: Intent = Intent(
            this@UniversityBaseDecToBaseActivity,
            UniversityBaseBaseToDecActivity::class.java
        )
        finish()
        startActivity(go)
    }

    private fun onRightSwipe() {
        val t: Toast =
            Toast.makeText(this@UniversityBaseDecToBaseActivity, "Right swipe", Toast.LENGTH_LONG)
        t.show()
        val go: Intent = Intent(
            this@UniversityBaseDecToBaseActivity,
            UniversityBaseBaseToDecActivity::class.java
        )
        finish()
        startActivity(go)
    }

    companion object {
        private const val SWIPE_MIN_DISTANCE = 50
        private const val SWIPE_MAX_OFF_PATH = 200
        private const val SWIPE_THRESHOLD_VELOCITY = 200
    }
}
